rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,90 @@
+---
+authors: Andrew LeFevre (andrew.lefevre@goteleport.com)
+state: draft
+---
+
+# RFD 98 - Registered OpenSSH Nodes
+
+## Required approvers
+
+* Engineering: @jakule && @r0mant
+* Product: @klizhentas
+* Security: @reedloden
+
+## What
+
+Allow OpenSSH nodes to be registered in a Teleport cluster.
+
+## Why
+
+[Agentless EC2 discovery mode](https://github.com/gravitational/teleport/issues/17865) will discover and configure OpenSSH nodes so they can authenticate with a cluster. But those OpenSSH nodes aren't registered as `node` resources in the backend. We need a way to register agentless OpenSSH nodes as `node` resources so they can be viewed and managed by users.
+
+## Details
+
+### Registering nodes
+
+OpenSSH nodes can already be registered using `tctl create -f /path/to/node.yml`, though some changes should be made to make the process more straightforward for users. Currently nodes cannot be created with `tctl`, only upserted. This limitation should be removed so users can create nodes without having to pass `-f` to `tctl create`.
+
+Furthermore, `tctl` should not require as many fields to be set when creating nodes. This is an example `node` resource that will work with `tctl create` today:
```

Comment:
You've outlined the problem here, but I don't see a solution.

Are you going to stick with `tctl create` and relax the required fields, or add a new command to register nodes that doesn't require dealing with YAML?

Reply:
Thanks, didn't realize I neglected to specify a solution.